langchain4j-azure-cosmos/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/mongo/vcore/AzureCosmosDbMongoVCoreEmbeddingStore.java

```diff
@@ -0,0 +1,622 @@
+            this.similarity = similarity;
+            return this;
+        }
+
+        /**
+         * @param dimensions - Number of dimensions for vector similarity. The maximum number of supported dimensions
+         *                   is 2000.
+         * @return
+         */
+        public Builder dimensions(Integer dimensions) {
+            this.dimensions = dimensions;
+            return this;
+        }
+
+        /**
+         * @param m - The max number of connections per layer (16 by default, minimum value is 2, maximum
+         *          value is 100). Higher m is suitable for datasets with high dimensionality and/or high
+         *          accuracy requirements.
+         * @return
+         */
+        public Builder m(Integer m) {
+            this.m = m;
+            return this;
+        }
+
+        /**
+         * @param efConstruction - the size of the dynamic candidate list for constructing the graph (64 by default, minimum
+         *                       value is 4, maximum value is 1000). Higher ef_construction will result in better index
+         *                       quality and higher accuracy, but it will also increase the time required to build the index.
+         *                       ef_construction has to be at least 2 * m.
+         * @return
+         */
+        public Builder efConstruction(Integer efConstruction) {
+            this.efConstruction = efConstruction;
+            return this;
+        }
+
+        /**
+         * @param efSearch - The size of the dynamic candidate list for search (40 by default). A higher value provides
+         *                 better recall at the cost of speed.
+         * @return
+         */
+        public Builder efSearch(Integer efSearch) {
+            this.efSearch = efSearch;
+            return this;
+        }
+
+        public AzureCosmosDbMongoVCoreEmbeddingStore build() {
+            return new AzureCosmosDbMongoVCoreEmbeddingStore(mongoClient, connectionString, databaseName, collectionName, indexName, applicationName,
+                    createCollectionOptions, createIndex, kind, numLists, similarity, dimensions, m,
+                    efConstruction, efSearch);
+        }
+    }
+
+    public enum AzureCosmosDBSimilarityType {
+        COS("COS"),
+        IP("IP"),
+        L2("L2");
+
+        private final String value;
+
+        AzureCosmosDBSimilarityType(String value) {
+            this.value = value;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+        public static AzureCosmosDBSimilarityType fromString(String kindString) {
+            return Arrays.stream(AzureCosmosDBSimilarityType.values())
+                    .filter(k -> k.getValue().equals(kindString))
+                    .findFirst()
+                    .orElse(AzureCosmosDBSimilarityType.COS); // Default value
+        }
+    }
+
+    public enum AzureCosmosDBVectorSearchType {
```

Comment:
```suggestion
    public enum VectorIndexType {
```
No need to duplicate "AzureCosmosDB" if it is an inner class

Reply:
Updated the name